src/rules/no-anonymous-default-export.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @fileoverview Rule to disallow anonymous default exports.
+ * @author Duncan Beevers
+ */
+
+const defs = {
+  ArrayExpression: {
+    option: 'allowArray',
+    message: 'an array',
+  },
+  ArrowFunctionExpression: {
+    option: 'allowArrowFunction',
+    message: 'an arrow function',
+  },
+  ClassDeclaration: {
+    option: 'allowAnonymousClass',
+    message: 'an anonymous class',
+    forbid: (node) => !node.declaration.id,
+  },
+  FunctionDeclaration: {
+    option: 'allowAnonymousFunction',
+    message: 'an anonymous function',
+    forbid: (node) => !node.declaration.id,
+  },
+  Literal: {
+    option: 'allowLiteral',
+    message: 'a literal',
+  },
+  ObjectExpression: {
+    option: 'allowObject',
+    message: 'an object expression',
+  },
+}
+
+const defaultOptions = Object.keys(defs).
+  map((key) => defs[key]).
+  reduce((acc, def) => {
+    acc[def.options] = false
+
+    return acc
+  })
+
+const schemaProperties = Object.keys(defs).
+  map((key) => defs[key]).
+  reduce((acc, def) => {
+    acc[def.option] = {
+      description: 'If `false`, will not report default export of ' + def.message,
```

Comment:
This sounds like the opposite of what the name suggests.
From what I understand, if `allowLiteral` is set to true, then Literals will be reported. :thinking: 

This should be reversed and the default value should be false.